cmd/lncli/walletrpc_active.go

```diff
@@ -477,6 +480,64 @@ func labelTransaction(ctx *cli.Context) error {
 	return nil
 }
 
+var publishTxCommand = cli.Command{
+	Name:      "publishtx",
+	Usage:     "Attempts to publish the passed transaction to the network.",
+	ArgsUsage: "tx_hex",
+	Description: `
+	Publish a base64-encoded raw transaction to the on-chain network. The 
+	wallet will continually attempt to re-broadcast the transaction on start up, until it 
+	enters the chain. The label parameter is optional and limited to 500 characters. Note 
+	that multi word labels must be contained in quotation marks ("").
+	`,
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:  "label",
+			Usage: "(optional) transaction label",
+		},
+	},
+	Action: actionDecorator(publishTransaction),
+}
+
+func publishTransaction(ctx *cli.Context) error {
+	ctxc := getContext()
+
+	// Display the command's help message if we do not have the expected
+	// number of arguments/flags.
+	if ctx.NArg() != 1 || ctx.NumFlags() > 1 {
+		return cli.ShowCommandHelp(ctx, "publishtx")
+	}
+
+	walletClient, cleanUp := getWalletClient(ctx)
+	defer cleanUp()
+
+	txBase64, err := base64.StdEncoding.DecodeString(ctx.Args().Get(0))
+	if err != nil {
+		return err
+	}
+
+	// Deserialize the transaction to get the transaction hash
+	tx := &wire.MsgTx{}
+	txReader := bytes.NewReader(txBase64)
+	if err := tx.Deserialize(txReader); err != nil {
+		return err
+	}
+
+	req := &walletrpc.Transaction{
+		TxHex: txBase64,
+		Label: ctx.String("label"),
+	}
+
+	_, err = walletClient.PublishTransaction(ctxc, req)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Transaction published (txid: %v)\n", tx.TxHash())
```

Comment:
I know this might not be true for all existing `walletrpc` commands, but we should return a JSON response here to make it more easily parseable by scripts. See `finalizePsbt` as an example.

Reply:
@guggero I'll see what I can do.

Reply:
Unfortunately that response doesn't include the TXID. I was thinking more along the lines of this:

```
	printJSON(&struct {
		TXID string `json:"txid"`
	}{
		TXID: hex.EncodeToString(tx.TxHash()),
	})
```

That would mean you'd need to parse the TX again, but that's probably a good idea anyway to avoid a round trip to the server if it's invalid.

Reply:
@guggero No problem. Updated the commit right now.